### PR TITLE
Pubdash: Adds docs for pubdash email sharing feature

### PR DESCRIPTION
**Pull request by owensmallwood**
fixes https://github.com/grafana/grafana-enterprise/issues/4709

Added docs for the email sharing feature. Goal was to cover the main workflows and touch on some of limitations (good and bad) of the feature. Decided to keep everything within the single public dashboards page since there isn't a ton of content already there, and the nav bar on the right makes it pretty easy to jump around already.

**Notes for reviewer**
Internally, we use the term "magic link" to describe one-time use links. I'm using the latter in the docs. Do we want to use "magic link" externally?

Screenshot 2023-03-09 at 2 51 53 PM

### CODE REVIEW

**Review by juanicabanas — APPROVED** [review]
Like it!! Let's wait until @imatwawana review though

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:67]:
```suggestion
## Email sharing
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:71]:
```suggestion
Email sharing allows you to share your public dashboard with only specific people via email, instead of having it accessible to anyone with the URL.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:75]:
```suggestion
> **Note:** For Grafana Cloud, contact support to have the feature enabled.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:84]:
```suggestion
### Invite a viewer
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:89]: How do you get to step 1 from the dashboard's normal view state? We're missing a couple preliminary steps here. 

```suggestion
1. Click **Only specified people**.
1. Enter the email you want to share the public dashboard with.
1. Click **Invite**.
1. The recipient(s) will receive an email with a one-time use link. This link must be used within **one hour** or it expires. Once the link is used, the viewer has access to the public dashboard for **30 days**.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:93]: What happens if they don't have an already invited email that hasn't been revoked? Is there an approval process?

```suggestion
If a viewer without access tries to navigate to the public dashboard, they will be asked to request access by providing their email. They'll receive an email with a new one-time use link if the email they provided has already been invited to view the public dashboard and has not been revoked.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:95]:
```suggestion
### Revoke access for a viewer
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:98]: A few things here:

- Same thing as above, we want to add in the preliminary steps. 
- Is "revoke" lower case in the UI? 
- I've also made a small change to avoid repetition. 
- I've removed step 2 from the steps because it's not a step the user takes, but an outcome of a steps they've taken.
- 

```suggestion
1. Click **revoke** on the viewer you'd like to remove access for.

Immediately, the user no longer has access to the public dashboard, nor can they use any existing one-time use links they may have.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:100]:
```suggestion
### Renew access for a viewer
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:103]: Add prelim steps
```suggestion
1. Click **Resend** on the viewer you'd like to re-share the public dashboard with.
1. The viewer will receive an email with a new one-time use link. This will invalidate all previously issued links for that viewer.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:107]:
```suggestion
One-time use links use browser cookies, so when a viewer is granted access via one of these links, they only have access through the browser they used to claim the link.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:109]:
```suggestion
A single viewer cannot generate multiple valid one-time use links. When a new one-time use link is issued for a user, all previous ones are invalidated.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:111]:
```suggestion
If a Grafana user has read access to the parent dashboard, they can view the public dashboard without needing to have access granted.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:113]:
```suggestion
## Supported data sources
```

**Review by imatwawana — COMMENTED** [review]
This is a great start! All of the tasks need to have the preliminary steps (how do you get to this set of buttons from just viewing a dashboard) added. Otherwise I've made a number of suggestions around wording/style/formatting and simplified the step numbering per markdown best practices. 

I'd like to take another look at this before it's merged, so please re-request review when you're ready.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:113]: With regard to "one-time use link" vs "magic link", whatever we use, it should be consistent in the error messages and in the docs. I'm kinda thinking that in the error messages, we could even drop the word "magic" from those messages and then it would line up with what we have in the docs. I don't feel strongly about which one we use (though I'll check with my team if there's any overall preference), as long as we use the term everywhere the user can see it.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:89]: Right, I'll add that. The steps for other pubdash actions above have it.

I was wondering why the number list was all 1's in other places. What's the point of that?

Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:89]: > Right, I'll add that. The steps for other pubdash actions above have it.
> 
> I was wondering why the number list was all 1's in other places. What's the point of that?

using the 1. method basically eliminates human error in numbering and allows you to really easily add in new steps anywhere in a task without having to re-number everything. basically it's like letting markdown do some formatting for you rather than doing it manually.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:113]: @Ijin08 - Can you weigh in on the "one-time use link" vs "magic link" question? Is there a UX preference in terms of how we talk about this in the app/docs?

Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:113]: Update: after some discussion, preference from docs team is "one-time use link" so no change to these docs, but I think we should update some of the error message text to match. Instead of "This magic link has expired" just "This link has expired." But will wait for @Ijin08 to add his thoughts.

Comment (evictorero) [reply docs/sources/dashboards/dashboard-public/index.md:113]: +1 to one-time link, the "magic" thing about magic link could be confusing.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (evictorero) [reply docs/sources/dashboards/dashboard-public/index.md:98]: Revoke and Resend are both uppercase
image

File: docs/sources/dashboards/dashboard-public/index.md
Comment (evictorero) [reply docs/sources/dashboards/dashboard-public/index.md:100]: I'm curious about why change `Reshare` by `Renew`, we don't use it in other places. I think `Renew` could imply something different.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (evictorero) [reply docs/sources/dashboards/dashboard-public/index.md:113]: We use mostly use `datasources` instead of `data sources`. Should we change all? Not sure how we are using this word in other doc places but in public dashboards at least it's always `datasources`.

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:100]: I was trying to avoid some awkward wording, but if "renew" could imply something different, then we should change it. Perhaps, since this is the word we use for first giving a person access:
```suggestion
### Reinvite a viewer
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:73]: I'm pretty sure this is covered earlier on the page [here](https://github.com/grafana/grafana/pull/64563/files#diff-3bc9469fa6aac8a43ec0ac831c6e29aed45b576173df43383fc8737431adac22L26)

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:113]: > We use mostly use `datasources` instead of `data sources`. Should we change all? Not sure how we are using this word in other doc places but in public dashboards at least it's always `datasources`.

Yeah, definitely change it to "data sources". It's two words in the application and in the [docs on the topic](https://grafana.com/docs/grafana/latest/datasources/).

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:109]: Going to change to `is issued for a viewer`. I think its important to differentiate between users and viewers

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:113]: > 

RIP all our code

Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:113]: > +1 to one-time link, the "magic" thing about magic link could be confusing.

RIP all our code

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:91]:
```suggestion
1. The recipient(s) will receive an email with a one-time use link. This link must be used within **one hour** or it expires. Once the link is used, the viewer has access to the public dashboard for **30 days**.
```

File: docs/sources/dashboards/dashboard-public/index.md
Comment (imatwawana) [inline docs/sources/dashboards/dashboard-public/index.md:95]: What happens to the request if you haven't already given them access?

**Review by imatwawana — APPROVED** [review]
I've left one question that I think we should probably answer before publishing, but otherwise LGTM!

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:95]: Nothing happens. They don't get an email or any notification that it hasn't been shared with them. This is intentional though. We don't want unwanted people trying to probe for which emails have access.

Maybe we should show a message after they request access saying "You'll receive an email with a link if you're allowed to view this dashboard", or something along those lines.

Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:95]: Does the dashboard owner get a notification of any kind?

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:73]: agree, I think it's implied already

File: docs/sources/dashboards/dashboard-public/index.md
Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:95]: Not at this point. It would be a nice feature to have, but wasn't in the cards for the V1 of this feature. I imagine it will happen later though.

Comment (owensmallwood) [reply docs/sources/dashboards/dashboard-public/index.md:95]: @evictorero @juanicabanas

Comment (imatwawana) [reply docs/sources/dashboards/dashboard-public/index.md:95]: It might be worth saying that explicitly.

**Pull request merged**
